packages/react/src/components/MultiSelect/__tests__/MultiSelect-test.js

```diff
@@ -3,380 +3,443 @@
+        />
       );
-      wrapper.find(`.${prefix}--list-box__field`).simulate('click');
-      const simulateArrowDown = () =>
-        wrapper.find(`.${prefix}--list-box__field`).simulate('keydown', {
-          key: 'ArrowDown',
-        });
-      const simulateArrowUp = () =>
-        wrapper.find(`.${prefix}--list-box__field`).simulate('keydown', {
-          key: 'ArrowUp',
-        });
-      const getHighlightedId = () =>
-        wrapper.find(`.${prefix}--list-box__menu-item--highlighted`).prop('id');
-      simulateArrowDown();
-      expect(getHighlightedId()).toBe('downshift-13-item-0');
-      simulateArrowDown();
-      expect(getHighlightedId()).toBe('downshift-13-item-1');
-      // Simulate "wrap" behavior
-      simulateArrowDown();
-      simulateArrowDown();
-      simulateArrowDown();
-      simulateArrowDown();
-      expect(getHighlightedId()).toBe('downshift-13-item-0');
-      simulateArrowUp();
-      expect(getHighlightedId()).toBe('downshift-13-item-4');
-    });
 
-    it('should close the menu when a user clicks outside of the control', () => {
-      const wrapper = mount(
-        <MultiSelect id="test-multiselect" {...mockProps} />
-      );
-      wrapper.find(`.${prefix}--list-box__field`).simulate('click');
-      mouseDownAndUp(document.body);
-      expect(wrapper.state('isOpen')).toBe(false);
-    });
+      const labelNode = getByText(container, label);
+      Simulate.click(labelNode);
 
-    it('should show a badge that mirrors the number of selected items', () => {
-      const wrapper = mount(
-        <MultiSelect id="test-multiselect" {...mockProps} />
-      );
-      openMenu(wrapper);
-      for (let i = 0; i < mockProps.items.length; i++) {
-        wrapper
-          .find(`.${prefix}--list-box__menu-item`)
-          .at(i)
-          .simulate('click');
-        expect(wrapper.find(`.${prefix}--list-box__selection`).text()).toEqual(
-          expect.stringContaining(`${i + 1}`)
-        );
-      }
-      expect(wrapper.find('Selection').instance().state.selectedItems).toEqual(
-        mockProps.items
-      );
-    });
+      const [item] = items;
+      const itemNode = getByText(container, item.label);
+      Simulate.click(itemNode);
 
-    it('should allow a user to de-select an item by clicking on a selected item', () => {
-      const wrapper = mount(
-        <MultiSelect id="test-multiselect" {...mockProps} />
-      );
-      wrapper.find(`.${prefix}--list-box__field`).simulate('click');
-      wrapper
-        .find(`.${prefix}--list-box__menu-item`)
-        .at(0)
-        .simulate('click');
-      expect(
-        wrapper.find(`.${prefix}--list-box__menu-item--active`).length
-      ).toBe(1);
-      wrapper
-        .find(`.${prefix}--list-box__menu-item`)
-        .at(0)
-        .simulate('click');
-      expect(
-        wrapper.find(`.${prefix}--list-box__menu-item--active`).length
-      ).toBe(0);
+      expect(testFunction).toHaveBeenCalled();
```

Comment:
This might be more of a style guide thing, but it's helpful to specify the expected number of calls. Sometimes tests can bleed details between them, or things are firing more often than we expect, and it's helpful to use `toHaveBeenCalledTimes` as a canary to indicate one of these problems might be occuring.

```suggestion
      expect(testFunction).toHaveBeenCalledTimes(1);
```